File: gluecodium/src/test/resources/smoke/annotations/output/android/com/example/smoke/Annotations.java
/*

 *
 */

package com.example.smoke;

import com.example.NativeBase;
import my.custom.annotations.NeverBeNull;
import my.other.annotations.Optional;

final class Annotations extends NativeBase {

    /**
     * For internal use only.
     * @hidden
     * @param nativeHandle The SDK nativeHandle instance.
     * @param dummy The SDK dummy instance.
     */
    protected Annotations(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }

    private static native void disposeNativeHandle(long nativeHandle);



    @Optional
    public native Boolean testOptional(@NeverBeNull final Annotations self);



}

File: gluecodium/src/test/resources/smoke/annotations/output/android-kotlin/com/example/smoke/Annotations.kt
/*

 *
 */

@file:JvmName("AnnotationsExtensions")
@file:androidx.annotation.OptIn(com.example.MySmokeTestsInternalApi::class)

package com.example.smoke

import com.example.MySmokeTestsInternalApi
import com.example.NativeBase

@MySmokeTestsInternalApi
internal class Annotations : NativeBase {



    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}





    @JvmSynthetic @JvmName("testOptional") internal external fun testOptional(self: Annotations) : Boolean?




    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}

